src/appleseed.studio/mainwindow/project/projectexplorer.h

```diff
@@ -84,17 +85,20 @@ class ProjectExplorer
     void clear_selection() const;
     ItemBase* select_entity(const foundation::UniqueID uid) const;
 
+    const MaterialDropHandler* get_material_drop_handler() const;
+
   signals:
     void signal_project_modified() const;
     void signal_frame_modified() const;
 
   private:
-    QTreeWidget*                m_tree_widget;
-    AttributeEditor*            m_attribute_editor;
-    ProjectBuilder              m_project_builder;
-    ItemRegistry                m_item_registry;
-    EntityEditorContext         m_editor_context;
-    std::unique_ptr<QShortcut>  m_delete_shortcut;
+    QTreeWidget*                            m_tree_widget;
+    AttributeEditor*                        m_attribute_editor;
+    ProjectBuilder                          m_project_builder;
+    ItemRegistry                            m_item_registry;
+    EntityEditorContext                     m_editor_context;
+    std::unique_ptr<QShortcut>              m_delete_shortcut;
+    std::shared_ptr<MaterialDropHandler>    m_material_drop_handler;
```

Comment:
Do we need a `std::shared_ptr<>` here? Or would a `std::unique_ptr<>` be enough?